.github/workflows/checks.yml

```diff
@@ -0,0 +1,191 @@
+name: Checks
+on:
+  push:
+    branches:
+      - '[0-9]+.[0-9]+.x'
+  pull_request_target:
+    branches:
+      - '[0-9]+.[0-9]+.x'
+    paths-ignore:
+      - doc/**
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    outputs:
+      isDefaultBranch: ${{ steps['info']['outputs']['is-default-branch'] }}
+    steps:
+      - name: Fetch repository information
+        id: info
+        run: |
+          DEFAULT_BRANCH=$(curl -s https://api.github.com/repos/syndesisio/syndesis |jq -r .default_branch)
+          IS_DEFAULT_BRANCH=$(test "refs/heads/${DEFAULT_BRANCH}" == "${GITHUB_REF}" && echo true || echo false)
+          echo "::set-output name=is-default-branch::${IS_DEFAULT_BRANCH}"
+  changes:
+    needs: setup
+    runs-on: ubuntu-latest
+    outputs:
+      backend: ${{ steps.filter.outputs.backend }}
+      images: ${{ steps.filter.outputs.images }}
+      operator: ${{ steps.filter.outputs.operator }}
+      ui: ${{ steps.filter.outputs.ui }}
+    steps:
+      - uses: actions/checkout@v2
+        if: needs.setup.outputs.isDefaultBranch == 'true'
```

Comment:
So if it is not the default branch... it does not run anything at all? Because the changes.outpus.* will be false if this is not run, right?